appserver/transaction/jts/src/main/java/com/sun/jts/CosTransactions/RegisteredSyncs.java

```diff
@@ -113,6 +112,15 @@
      */
     RegisteredSyncs() {}
 
+    private RegisteredSyncs(List<Synchronization> rhs) {
+        registered.addAll(rhs);
+    }
+    
+    @Override
+    protected RegisteredSyncs clone() throws CloneNotSupportedException {
```

Comment:
Didn't test the MDB/MQ scenario. Thanks @pdudits !
As far as the CloneNotSupported exception, I get a warning otherwise, and it's good form to support it since it overrides the interface...